### PR TITLE
feat(@kampus/pano-button & @kampus/pano): Create an embeddable share component

### LINKED ISSUE #310: share button for kampus/pano to be used in another websites
We want to create an embedable "share this on pano.kamp.us" type of button, so people can use this in anywhere to post their content to pano.

### Things
- [ ] create a new package under `packages` folder named: `pano-button`
- [ ] add support for 

WIP

### PR DESCRIPTION
**Pull request by andyanday33**
# Description

I have created a new package containing a share button, that could be embedded in websites to point out to send route of pano with a prefilled URL value.
I have also added a query parameter to pano "/send" route, that is being checked for prefill the URL and title fields

### Checklist

- [X] discord username: `bumshaqalaqa#2973`
- [X] Closes #310 
- [X] PR must be created for an issue from issues under "In progress" column from [our project board](https://github.com/orgs/kamp-us/projects/2/views/1).
- [X] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules, and should include the app/package/service name in the title. For example, a title like "docs(@kampus-apps/pano): Add README.md" can be used.
- [X] Related file selection: Only relevant files should be touched and no other files should be affected.
- [X] I ran `npx turbo run` at the root of the repository, and build was successful.
- [X] I installed the npm packages using `npm install --save-exact <package>` so my package is pinned to a specific npm version. Leave empty if no package was installed. Leave empty if no package was installed with this PR.

### How were these changes tested?

I have added a temporary share button to the settings page, imported it from @kampus/pano-button package, and provided the component a URL value, then tested it manually.

### CODE REVIEW

File: packages/pano-button/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# `@kampus/pano-button`
+
+Embeddable share button for kampus apps.
```
Comment (usirin) [inline packages/pano-button/README.md:3]: [NIT] for pano, not kampus apps 😅

File: apps/pano/app/routes/send.tsx [hunk 2]
```diff
@@ -81,6 +83,13 @@ const Send = () => {
     }
   };
 
+  const [searchParams] = useSearchParams();
+  const queryUrl = searchParams.get("url");
+
+  useEffect(() => {
+    if (queryUrl) fetchMetaTags(queryUrl);
```
Comment (usirin) [inline apps/pano/app/routes/send.tsx:90]: let's move this to `loader` and return it from there, that way we can ensure that the values are actually generated on the server. then component can use the loader data as inputs' default values.

File: packages/pano-button/src/ShareButton.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  const handleClick = () => {
+    const shareUrl = `http://pano-dev.kamp.us/send?url=${encodeURIComponent(
+      url
+    )}`;
+
+    if (typeof window === "undefined") return;
+    window.open(shareUrl, "_blank");
+  };
+
+  return <Button onClick={handleClick}>Share this on kamp-us pano</Button>;
```
Comment (usirin) [inline packages/pano-button/src/ShareButton.tsx:19]: Let's use turkish here.

**Review by usirin — CHANGES_REQUESTED** [review]

**Comment by usirin** [comment]
@andyanday33 ping pong

**Comment by andyanday33** [comment]
> @andyanday33 ping pong

sorry, gonna work on this tomorrow.

**Comment by andyanday33** [comment]
Updated regarding changes suggested

**Comment by andyanday33** [comment]
currently, the default value of title input does not change on paste or blur when the query parameter "url" exists for some reason, it requires further investigation.

**Review by usirin — APPROVED** [review]

**Pull request merged**